common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1162 @@
+ */
+fs.mkdtemp = (prefix, options, callback) => {
+	assertArgumentType(prefix, 'prefix', 'string');
+	if (typeof options === 'function') {
+		callback = options;
+		options = {};
+	}
+	options = mergeDefaultOptions(options, { encoding: 'utf-8' });
+
+	// try to be all async
+	const tryMkdtemp = () => {
+		// FIXME: use the encoding specified!
+		const generated = randomCharacters(6); // generate six random characters
+		const path = `${prefix}${generated}`;
+		fs.mkdir(path, 0o700, err => {
+			if (err) {
+				if (err.code === 'EEXIST') {
+					// retry!
+					setTimeout(tryMkdtemp, 1);
+					return;
+				}
+				// bubble up error
+				callback(err);
+				return;
+			}
+			// succeeded! Hurray!
+			callback(null, path);
+		});
+	};
+	setTimeout(tryMkdtemp, 1);
+};
+
+/**
+ * Creates a unique temporary directory.
+ * @param {string} prefix directory name prefix
+ * @param {string|object} [options] options
+ * @param {string} [options.encoding='utf-8'] prefix encoding
+ * @returns {string} path to created directory
+ */
+fs.mkdtempSync = (prefix, options) => {
+	assertArgumentType(prefix, 'prefix', 'string');
+	console.log(`mkdtempSync - options type: ${typeof options}`);
+	options = mergeDefaultOptions(options, { encoding: 'utf-8' });
+
+	let retryCount = 0;
+	const MAX_RETRIES = 100;
+	while (retryCount < MAX_RETRIES) {
+		// FIXME: use the encoding specified!
+		const generated = randomCharacters(6); // generate six random characters
+		const path = `${prefix}${generated}`;
+		try {
+			fs.mkdirSync(path, 0o700); // don't try recursive
+			return path;
+		} catch (e) {
+			if (e.code !== 'EEXIST') {
+				throw e; // bubble up error
+			}
+			// name was not unique, so retry
+			retryCount++;
+		}
+	}
+	throw new Error(`Failed to create a unique directory name with prefix ${prefix}`);
+};
+
+/**
+ * @callback fileDescriptorCallback
+ * @param {Error} err - Error if one occurred
+ * @param {integer} fileDescriptor - generated file descriptor
+ */
+
+/**
+ * @param {string|Buffer|URL} path path to file
+ * @param {string} [flags='r'] file system access flags
+ * @param {integer} [mode=0o666] file mode to use when creating file
+ * @param {fileDescriptorCallback} callback async callback
+ */
+fs.open = (path, flags, mode, callback) => {
+	setTimeout(() => {
+		// flags and mode are optional, we need to handle if not supplied!
+		if (typeof flags === 'function') {
+			callback = flags;
+			flags = 'r';
+			mode = 0o666;
+		} else if (typeof mode === 'function') {
+			callback = mode;
+			mode = 0o666;
+		}
+
+		let fileDescriptor;
+		try {
+			fileDescriptor = fs.openSync(path, flags, mode);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback(null, fileDescriptor);
+	}, 1);
+};
+
+/**
+ * @param {string|Buffer|URL} path path to file
+ * @param {string} [flags='r'] file system access flags
+ * @param {integer} [mode=0o666] file mode to use when creating file
+ * @returns {integer}
+ */
+fs.openSync = (path, flags = 'r', mode = 0o666) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 592;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 592 – 'mode' is assigned a value but never used. (no-unused-vars)